conntrack_test.go

```diff
@@ -44,6 +45,57 @@ func udpFlowCreateProg(t *testing.T, flows, srcPort int, dstIP string, dstPort i
 	}
 }
 
+// Install minimal hooks so packets traverse conntrack in this netns.
+// Prefer iptables if available; otherwise use nftables.
+func ensureCtHooksInThisNS(t *testing.T) {
+	t.Helper()
+
+	// Prefer iptables if present
+	if _, err := exec.LookPath("iptables"); err == nil {
+		iptExists := func(args ...string) bool {
+			cmd := exec.Command("iptables", args...)
+			if out, err := cmd.CombinedOutput(); err == nil {
+				return true
+			} else {
+				// iptables -C returns non-zero if rule doesn't exist; that's OK.
+				t.Logf("iptables %v -> not present (ok): %v\n%s", args, err, out)
+				return false
+			}
+		}
+		ipt := func(args ...string) {
+			cmd := exec.Command("iptables", args...)
+			if out, err := cmd.CombinedOutput(); err != nil {
+				t.Fatalf("iptables %v failed: %v\n%s", args, err, out)
+			}
+		}
+
+		// Minimal hooks so packets traverse conntrack in this netns.
+		// Check (-C); if absent, insert (-I). Idempotent on reruns.
+		if !iptExists("-C", "INPUT", "-m", "conntrack", "--ctstate", "NEW,ESTABLISHED", "-j", "ACCEPT") {
```

Comment:
Given we are polluting the host with these rules, it would be better to add them to a custom iptables chain, and something that reconnects to this library UT
```
iptables -D INPUT -j vish_netlink
iptables -X vish_netlink
iptables -N vish_netlink
iptables -I vish_netlink -m conntrack --ctstate NEW, ESTABLISHED - j ACCEPT
```
and at test end we should cleanup
```
iptables -D INPUT -j vish_netlink
iptables -X vish_netlink
```

similarly for the nft table.

Reply:
@aboch The rules will only be created in the network namespace used for the test so they will be gone when the namespace is dropped. I can add a cleanup step to remove the rules for safety but I’m not sure the value of using a separate table.

Reply:
I modified the test to cleanup the rules it adds.

Reply:
Oh, of course Sorry I missed the setup would only be executed inside a netns. Please disregard my comment.